pkg/archive/pypi.go

```diff
@@ -0,0 +1,748 @@
+		case "Provides-Dist":
+			info.ProvidesDist = append(info.ProvidesDist, value)
+		case "Obsoletes-Dist":
+			info.ObsoletesDist = append(info.ObsoletesDist, value)
+		default:
+			info.OtherFields[key] = append(info.OtherFields[key], value)
+		}
+	}
+	return info, nil
+}
+
+// Helper function to detect structured information
+func isStructuredInfo(line []byte) bool {
+	// Check for patterns like links, images, or specific formats
+	if bytes.HasPrefix(line, []byte(".. image::")) ||
+		bytes.HasPrefix(line, []byte(".. _")) ||
+		bytes.Contains(line, []byte("http://")) ||
+		bytes.Contains(line, []byte("https://")) {
+		return true
+	}
+	return false
+}
+
+// func ParseRecordDistInfo(r io.Reader) (*RecordDistInfo, error) {
+// 	content, err := io.ReadAll(r)
+// 	if err != nil {
+// 		return nil, err
+// 	}
+// 	lines := bytes.Split(content, []byte("\n"))
+// 	record := &RecordDistInfo{
+// 		entries: make([]RecordDistEntry, 0),
+// 	}
+// 	for _, line := range lines {
+// 		if len(line) == 0 {
+// 			continue
+// 		}
+// 		// Parse the line into RecordDistEntry fields
+// 		parts := bytes.SplitN(line, []byte{','}, 3)
+// 		if len(parts) < 3 {
+// 			continue
+// 		}
+// 		size := int64(0)
+// 		if len(parts[1]) > 0 {
+// 			// Try to parse size
+// 			var err error
+// 			size, err = parseInt64(string(parts[1]))
+// 			if err != nil {
+// 				size = 0
+// 			}
+// 		}
+// 		entry := RecordDistEntry{
+// 			Path:   string(parts[0]),
+// 			Size:   size,
+// 			SHA256: string(parts[2]),
+// 		}
+// 		record.entries = append(record.entries, entry)
+// 	}
+// 	return record, nil
+// }
+
+var StablePypiMetadata = ZipArchiveStabilizer{
+	Name: "pypi-metadata",
+	Func: func(zr *MutableZipReader) {
+		for _, zf := range zr.File {
+			// Only process metadata.json files
+			if !strings.HasSuffix(zf.Name, "metadata.json") {
+				continue
+			}
+			println("Processing file:", zf.Name)
+
+			zf.SetContent([]byte("This needed to change (metadata)"))
+
+		}
+	},
+}
+
+var StablePypiDescription = ZipArchiveStabilizer{
+	Name: "pypi-description",
+	Func: func(zr *MutableZipReader) {
+		for _, zf := range zr.File {
+			// Only process METADATA files
+			if !strings.HasSuffix(zf.Name, "DESCRIPTION.rst") {
+				continue
+			}
+			println("Processing file:", zf.Name)
+			// rebuild the zip without the Description.rst file
+			// as it is not needed for stabilization.
+
+			zf.SetContent([]byte("This needed to change (description)"))
+
+		}
+	},
+}
+
+var StableWheelBuildMetadata = ZipEntryStabilizer{
+	Name: "whl-build-metadata",
+	Func: func(zf *MutableZipFile) {
+		// Only process METADATA files
+		if !strings.HasSuffix(zf.Name, "dist-info/METADATA") {
+			return
+		}
+		println("Processing file:", zf.Name)
+
+		// Open the existing METADATA file
+		r, err := zf.Open()
+		if err != nil {
+			println("Error opening METADATA file:", err)
+			return
+		}
+
+		// Parse the METADATA file
+		manifest, err := ParseMetadataDistInfo(r)
```

Comment:
hmm so actually per my comment above, you may not even need the structured golang object: You could just extract the 3-4 keys below that we special-case, and then unpack the rest naively from the mail.Header map.